packages/wasm-utils/src/note/anchor.rs

```diff
@@ -42,15 +41,31 @@ pub fn get_leaf_with_private_raw(
 		return Err(OpStatusCode::InvalidNoteSecrets.into());
 	}
 
-	let secrets = raw[..32].to_vec();
-	let nullifer = raw[32..64].to_vec();
+	let secrets;
+	let nullifier;
+	if raw.len() == 70 {
+		// 70 bytes raw implies [6 bytes chain ID, 32 bytes nullifier, 32 bytes secret]
+		nullifier = raw[6..38].to_vec();
+		secrets = raw[38..70].to_vec();
+	} else if raw.len() == 72 {
+		// 72 bytes raw implies [8 bytes chain ID, 32 bytes nullifier, 32 bytes secret]
+		nullifier = raw[8..40].to_vec();
+		secrets = raw[40..72].to_vec();
+	} else if raw.len() >= 64 {
+		// 64 bytes raw implies [32 bytes nullifier, 32 bytes secret]
+		secrets = raw[0..32].to_vec();
+		nullifier = raw[32..64].to_vec();
+	} else {
+		return Err(OpStatusCode::InvalidNoteSecrets.into());
+	}
+
 	// (leaf_bytes, nullifier_hash_bytes)
```

Comment:
`JsNote::get_leaf_and_nullifier` this function has the secrets already , for v1 notes the secrets are concatenated in a single entry `0..32` secret , `32...63` nullifier
and for v2 we already have the secrets` [chain_id,nullifier_bytes,secret_bytes]`

so I support changing the signature of `get_leaf_with_private_raw` to receive hain_id,nullifier_bytes,secret_bytes as parameters instead of receiving the secrets as `row:&[u8]`.
as we have all secrets splitted for v1 note